Nodejs/Product/Telemetry/Shared/ITelemetryLogger.cs

```diff
@@ -0,0 +1,55 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+
+namespace Microsoft.NodejsTools.Telemetry {
+    public interface ITelemetryLogger : IDisposable {

```

Comment:
I want the actual implementations of ITelemetryLogger to be notified when app exits. I have to find right place to call it from but didn't want to block checkin on it. I'll add that in a later checkin.
Dispose implementation is empty right now in this PR as it is a dummy logger. Other actual logger in internal code might do some cleanup like flushing data at end. If it makes more sense I add it later when it is really needed by some logger implementation then I can do that but I thought it would be better not to change shared interfaces much.